src/lib.rs

```diff
@@ -144,6 +144,8 @@ pub mod nightly {
 ///
 /// [std-overflow]: https://doc.rust-lang.org/reference/expressions/operator-expr.html#overflow
 #[derive(Clone, Copy, Eq, PartialEq, Hash)]
+#[cfg_attr(feature = "diesel", derive(diesel::AsExpression, diesel::FromSqlRow))]
+#[cfg_attr(feature = "diesel", diesel(sql_type = diesel::sql_types::Binary))]
```

Comment:
I would prefer not to have these leak into the codebase, is it possible to expand these and include them in the `support/diesel.rs` file? How many LoC would that add?

Reply:
Just pushed up these changes! 😁

It adds ~125 LoC (see latest commit):
<img width="139" alt="image" src="https://github.com/user-attachments/assets/d80c078e-5373-4c36-860a-4af439940574">

Tested with my demo repo against a fresh DB and regenerating migrations to sanity check! 🫡

Reply:
cool. I will review, and may do a style pass on the generated code if you don't mind

Reply:
Yes of course, feel free-- I mostly kept it the same as the generated code, minus deduping some long namespaced variables. Lmk if I can adjust anything else! 🫡

Thanks James 😁